### PR TITLE
Improve api docs cosmos db

### PR DESCRIPTION
**Pull request by tangerinous**
# Description

The PR adds additional information to the java docs according to the [issue](https://github.com/Azure/azure-sdk-for-java/issues/19815)

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
+package com.azure.cosmos;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java:1]: missing new file license header

File: sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+import com.azure.cosmos.models.*;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ContainerAsyncCodeSnippets {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java:9]:
```suggestion
public class AsyncContainerCodeSnippets {
```

File: sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    private final String serviceEndpoint = "<service-endpoint>";
+    private final String key = "<key>";
+    private final DirectConnectionConfig directConnectionConfig = new DirectConnectionConfig();
+    private final GatewayConnectionConfig gatewayConnectionConfig = new GatewayConnectionConfig();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java:13]: looks like these connection configs are not used.

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java (hunk 3 above)
Comment (tangerinous) [reply sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ContainerAsyncCodeSnippets.java]: Fixed it

**Pull request merged**